### ISSUE TITLE
not only BEM

### ISSUE THREAD

**Issue by max-norin** [future-architect/eslint-plugin-vue-scoped-css#48]
`vue-scoped-css/no-unused-selector` rule has `ignoreBEMModifier` option 
Besides `BEM`, there are other ways to organize CSS code. For example [RSCCS](https://rscss.io), where the modifier is defined as in the following code. More details [here](https://rscss.io/variants.html).
```
.like-button {
  &.-wide { /* ... */ }
  &.-short { /* ... */ }
  &.-disabled { /* ... */ }
}
```
Also, development teams can use custom modifier naming. Сreator of the [stylelint-selector-bem-pattern](https://github.com/simonsmith/stylelint-selector-bem-pattern) made it possible to specify a regular expression to define a modifier.  Maybe you should pay attention to this.

**Comment by ota-meshi** [comment]
Thank you for this issue!

I am not familiar with RSCCS.
Can you write a detailed specification of the features you want to add, in this issue?
Also, can you open a pull request after that?

**Comment by max-norin** [comment]
For example,
replace `ignoreBEMModifier` option with:
```js
ignoreModifier: {
   // The preset patterns are available: bem and rscc
   preset: 'bem'|'rscc',
   // Array of regular expressions or strings defining custom modifier naming
   list: [
     '/{componentSelector}--{modifier}/', // this regular expression match bem 
     '/{componentSelector}\\.-{modifier}/', // this regular expression match rscc
     '&--warn', '&--error', // this strings match bem where as modifiers there can only be `warn` and `error`
     '&.-warn', '&.-error', // this strings match rscc where as modifiers there can only be `warn` and `error`
     '/{componentSelector}-{modifier}/', // this regular expression match custom modifier naming
     '/{componentSelector}---{modifier}/', // this regular expression match custom modifier naming
     '/{componentSelector}.{modifier}/', // this regular expression match custom modifier naming
   ]
}
```
